artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -152,11 +152,54 @@ async def search_builds_by_fields(self, start_search: datetime, end_search: date
         self.logger.debug('Found %s builds', results.total_rows)
         return results
 
+    async def search_builds_by_fields_with_window(
```

Comment:
fine with me, but I think we could just have one single `search_builds_by_fields()` function that always scans within a moving window of fixed size. `start_time` should be optional, and if None just assume a very old date (you used 3 years, that can be adjusted if we want)

Reply:
Done merging into one single `search_builds_by_fields`